src/components/SignalGeneratorComponent.cpp

```diff
@@ -0,0 +1,125 @@
+#include <iostream>
+#include <hyro/utils/SpinnerDefault.h>
+#include <hyro/factory/CommandFactory.h>
+#include <hyro/SignalGeneratorComponent.h>
+#include <hyro/SignalGenerator.h>
+#include <hyro/msgs/Signal.h>
+#include <hyro/utils/DynamicPropertyAccess.h>
+
+using namespace std::chrono_literals;
+
+namespace hyro
+{
+std::shared_ptr<HyroLogger> SignalGeneratorComponent::s_logger = HyroLoggerManager::CreateLogger("SignalGeneratorComponent");
+std::shared_ptr<ChannelOutput<Signal>> m_signal_output;
+hyro::DynamicPropertyAccess dynamic_property_access_signal("/signal"_uri);
+SignalGenerator sg;
+
+Result
+SignalGeneratorComponent::init (const ComponentConfiguration & configuration)
+{
+  registerDynamicProperty<double>("amplitude",
+                                  &SignalGeneratorComponent::setAmplitude,
+                                  &SignalGeneratorComponent::getAmplitude,
+                                  this
+                                  );
+
+  registerDynamicProperty<double>("frequency",
+                                  &SignalGeneratorComponent::setFrequency,
+                                  &SignalGeneratorComponent::getFrequency,
+                                  this
+                                  );
+
+  registerDynamicProperty<bool>("cosine",
+                                &SignalGeneratorComponent::setCosine,
+                                &SignalGeneratorComponent::getCosine,
+                                this
+                                );
+
+  std::shared_ptr<ChannelOutput<std::vector<int>>>
+  m_dummy = this->registerOutput<std::vector<int>>("fix_dynamic"_uri, configuration);
+  
+  m_output = this->registerOutput<Signal>("value"_uri, configuration);
+
+  // Always return a result code.
+  return Result::RESULT_OK;
+}
+
+double
+SignalGeneratorComponent::getAmplitude()
+{
+  return amplitude;
+}
+
+bool 
+SignalGeneratorComponent::setAmplitude(double ampl) 
+{ 
+  amplitude = ampl; 
+  sg.setAmplitude(amplitude);
+  return true;
+}
+
+double
+SignalGeneratorComponent::getFrequency()
+{
+  return frequency;
+}
+
+bool 
+SignalGeneratorComponent::setFrequency(double freq) 
+{ 
+  frequency = freq; 
+  sg.setFrequency(frequency);
+  return true;
+}
+
+bool
+SignalGeneratorComponent::getCosine()
+{
+  return cosine;
+}
+
+bool 
+SignalGeneratorComponent::setCosine(bool coss) 
+{ 
+  cosine = coss; 
+  sg.setCosine(cosine);
+  return true;
+}
+
+Result
+SignalGeneratorComponent::check ()
+{ 
+  return Result::RESULT_OK;
+}
+
+Result
+SignalGeneratorComponent::start ()
+{
+  double freq = 1;
```

Comment:
This initialization goes in the `init` method, as @gustavoneves12  said. 
And avoid to hard-coded the parameters like that. In the `init` you can get the parameters from the configuration, with a default value.

```
double frequency = configuration.parameters.getParameter<float>("frequency", 1);
sg.setFrequency(frequency);
```